### DISCUSSION TITLE
Add a README for each major module

### DISCUSSION THREAD

**Discussion by lrzpellegrini** [ContinualAI/avalanche#392] [Ideas]
Hello everyone, I noticed that many big projects use a module-specific README.md to better explain the features provided by each module (and to provide extended examples).

This kind of README usually contains a very high-level description of the content of that module (and its sub-modules), should have a proper list of references to the related (gitbook) tutorials and finally give some examples ranging from "newbie" to "advanced".

I think that we should add at least a README for each of the major sub-modules: benchmarks, training, logging and evaluation. Adding a short README for the "models" module may be a good idea, too.

What do you think of that?

**Comment by vlomonaco** [comment]
This is a nice idea. I would have them as "shallow" as possible though (just links, mostly) to maintain the doc only on gitbook and the Avalanche API doc!

**Comment by AntonioCarta** [comment]
Is this common? I've never seen it in large python projects. Often, they only have a very high level description (couple of sentences) in `__init__.py`.

**Comment by jeremyforest** [comment]
Wouldn't that be a duplicate of the doc though ? While adding to the maintenance work ?

**Comment by vlomonaco** [comment]
Moving this issue in discussions till we agree if we really need it.
